packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -2315,6 +2316,201 @@ class ConstrainedBox extends SingleChildRenderObjectWidget {
   }
 }
 
+/// A container widget that applies an arbitrary transform to its constraints,
+/// and sizes its child using the resulting [BoxConstraints], treating any
+/// overflow as error.
+///
+/// This container sizes its child using a [BoxConstraints] created by applying
+/// [constraintsTransform] to its own constraints. This container will then
+/// attempt to adopt the same size, within the limits of its own constraints. If
+/// it ends up with a different size, it will align the child based on
+/// [alignment]. If the container cannot expand enough to accommodate the entire
+/// child, the child will be clipped if [clipBehavior] is not [Clip.none].
+///
+/// In debug mode, if the child overflows the container, a warning will be
+/// printed on the console, and black and yellow striped areas will appear where
+/// the overflow occurs.
+///
+/// {@tool snippet}
+/// In the following snippet, the child [Card] is guaranteed to be at least as
+/// tall as its intrinsic height. Unlike [UnconstrainedBox], the child can be
+/// taller if the the parent's minHeight is higher than the child's intrinsic
+/// height. If parent container's maxHeight is less than [Card]'s intrinsic
+/// height, in debug mode a warning will be given.
+///
+/// ```dart
+/// ConstraintsTransformBox(
+///   constraintsTransform: (BoxConstraints constraints) => constraints.copyWith(maxHeight: double.infinity),
+///   child: const Card(child: Text('Hello World!')),
+/// )
```

Comment:
Could this also be achieved with the code below, or am I misunderstanding?

```dart
LayoutBuilder(
  builder: (BuildContext context, BoxConstraints constraints) {
    return ConstrainedBox(
      constraints: constraints.copyWith(maxHeight: double.infinity),
      child: const Card(child: Text('Hello World!')),
    );
  },
)

Reply:
`ConstrainedBox` is a bit different from `ConstraintsTransformBox`, as it does not replace the original constraints but adds additional constraints to the original constraints. In this example, if the parent is a `SizedBox(height: 50)` then in the above example the Card can be as high as it wants but with `ConstrainedBox` the maxium height is 50. I'll update the example to be more specific.

Reply:
I really thought that my example here was the same as ConstraintsTransformBox, but I tried it and they are not the same.

So here's a simple example not using anything new:

```dart
            Container(
              height: 100,
              color: Colors.red,
              clipBehavior: Clip.none,
              child: ConstrainedBox(
                constraints: BoxConstraints(maxHeight: double.infinity),
                child: Container(
                  height: 200,
                  color: Colors.blue.withOpacity(0.5),
                ),
              ),
            ),
```

Both Containers are 100px high there.  There is no overflow.  I would have expected the blue Container to extend outside of the red Container, but I guess I'm wrong about how Flutter does this :)

<img width="801" alt="Screen Shot 2021-03-16 at 12 10 06 PM" src="https://user-images.githubusercontent.com/389558/111366388-a6ddee00-8650-11eb-8aa1-dea296756447.png">


Now here's a similar thing using ConstraintsTransformBox:

```dart
            Container(
              height: 100,
              color: Colors.yellow.withOpacity(0.5),
              child: ConstraintsTransformBox(
                alignment:Alignment.topLeft,
                constraintsTransform: (BoxConstraints constraints) => constraints.copyWith(maxHeight: double.infinity),
                child: Container(
                  height: 200,
                  color: Colors.blue.withOpacity(0.5),
                ),
              ),
            ),
```

The inner blue Container is able to extend outside of its parent, with an overflow warning.

<img width="800" alt="Screen Shot 2021-03-16 at 12 10 32 PM" src="https://user-images.githubusercontent.com/389558/111366396-aa717500-8650-11eb-8626-e53cd7973ef4.png">

So I think this makes sense and I understand how this widget produces a unique behavior that doesn't seem to be easily doable with existing widgets 👍